contracts/src/v0.8/dev/ArbitrumValidator.sol

```diff
@@ -0,0 +1,140 @@
+
+  ArbitrumInboxInterface private s_arbitrumInbox;
+  SimpleWriteAccessController private s_gasConfigAccessController;
+
+  struct GasConfiguration {
+    uint256 maximumSubmissionCost;
+    uint32 maximumGasPrice;
+    uint256 gasCostL2;
+    address refundableAddress;
+  }
+  GasConfiguration private s_gasConfig;
+  uint32 constant private s_L2GasLimit = 30000000;
+  uint32 constant private s_maxSubmissionCostIncreaseRatio = 13;
+
+  /**
+   * @param aggregatorAddress default aggregator with access to validate
+   * @param inboxAddress address of the Arbitrum Inbox L1 contract
+   * @param flagAddress address of the Chainlink L2 Flags contract
+   * @param gasConfigAccessController address of the access controller for managing gas price on Arbitrum
+   * @param maxSubmissionCost maximum cost willing to pay on L2
+   * @param maximumGasPrice maximum gas price to pay on L2
+   * @param gasCostL2 value to send to L2 to cover gas fee
+   * @param refundableAddress address where gas excess on L2 will be sent
+   */
+  constructor(
+    address aggregatorAddress,
+    address inboxAddress,
+    address flagAddress,
+    address gasConfigAccessController,
+    uint256 maxSubmissionCost,
+    uint32 maximumGasPrice,
+    uint256 gasCostL2,
+    address refundableAddress
+  ) {
+    s_arbitrumInbox = ArbitrumInboxInterface(inboxAddress);
+    s_gasConfigAccessController = SimpleWriteAccessController(gasConfigAccessController);
+    s_flagsAddress = flagAddress;
+    _setGasConfiguration(maxSubmissionCost, maximumGasPrice, gasCostL2, refundableAddress);
+
+    SimpleWriteAccessController(address(this)).addAccess(aggregatorAddress);
+  }
+  
+  fallback() external payable {}
+
+  function withdrawFunds() 
+    external 
+    onlyOwner() 
+  {
+    address payable to = payable(msg.sender);
+    to.transfer(address(this).balance);
+  }
+
+  function withdrawFundsTo(
+    address payable to
+  ) 
+    external
+    onlyOwner() 
+  {
+    to.transfer(address(this).balance);
+  }
+
+  function setGasConfiguration(
+    uint256 maxSubmissionCost,
+    uint32 maximumGasPrice,
+    uint256 gasCostL2,
+    address refundableAddress
+  )
+    external
+  {
+    require(s_gasConfigAccessController.hasAccess(msg.sender, msg.data), "Only gas configuration admin can call");
+    _setGasConfiguration(maxSubmissionCost, maximumGasPrice, gasCostL2, refundableAddress);
+  }
+
+  function validate(
+    uint256 /* previousRoundId */,
+    int256 /* previousAnswer */,
+    uint256 /* currentRoundId */,
+    int256 currentAnswer
+  ) 
+    external 
+    override
+    checkAccess() 
+    returns (bool) 
+  {
+    // Avoids resending to L2 the same tx on every call
+    if (s_lastChangedAnswer == currentAnswer) {
+      return true;
+    }
+    s_lastChangedAnswer == currentAnswer;
+    bytes memory data = currentAnswer == 1 ? abi.encodeWithSelector(RAISE_SELECTOR, s_arbitrumFlag) : abi.encodeWithSelector(LOWER_SELECTOR, [s_arbitrumFlag]);
```

Comment:
What will the answer mean? I could see:
- 0 / 1 = Sequencer offline (raise flag) / Sequencer online (lower flag)
- 0 / 1 = Don't raise Flag (seq online) / Do Raise Flag (seq online)

Reply:
We named the flag `chainlink.flags.arbitrum-offline` so it suggests that when raised (1/true) this means Arbitrum is offline. I suggest we make the L1 feed and L2 flags referencing the same data point: **Is Arbitrum Sequencer offline? 1 -> true, 0 -> false**

We expect the Sequencer to mostly be online, meaning `chainlink.flags.arbitrum-offline = 0/false`, which will cost less in calldata and storage costs, than doing it the other way around.

Reply:
This was implemented here: https://github.com/smartcontractkit/chainlink/pull/4641/commits/ec4adf8284b45019b76fa578a01b4368e5d21f99

Also, the flag was renamed as `chainlink.flags.arbitrum-seq-offline` to be more explicit in what is exactly offline.